webui/react/src/shared/themes.test.ts

```diff
@@ -0,0 +1,86 @@
+import * as themes from './themes';
+import { isColor, labDistance, rgb2lab, str2rgba } from './utils/color';
+
+const supportedThemes = {
+  darkDet: themes.themeDarkDetermined,
+  darkHpe: themes.themeDarkHpe,
+  lightDet: themes.themeLightDetermined,
+  lightHpe: themes.themeLightHpe,
+};
+
+type ThemeVarPair = (keyof themes.Theme)[];
+const genPairs = (name: string): ThemeVarPair[] => {
+  const base = [
+    [ `${name}`, `${name}On` ],
+    [ `${name}`, `${name}Border` ],
+  ] as ThemeVarPair[];
+
+  // same intentsity pairs.
+  const strong = [ ...base.map(([ a, b ]) => [ `${a}Strong`, `${b}Strong` ]) ] as ThemeVarPair[];
+  const weak = [ ...base.map(([ a, b ]) => [ `${a}Weak`, `${b}Weak` ]) ] as ThemeVarPair[];
+
+  // stronger color on top of weaker color.
+  const strongOnDefault = [ ...base.map(([ a, b ]) => [ `${a}`, `${b}Strong` ]) ] as ThemeVarPair[];
+  const defaultOnWeak = [ ...base.map(([ a, b ]) => [ `${a}Weak`, `${b}` ]) ] as ThemeVarPair[];
+
+  // weak colors on top of strong colors.
+  // const weakOnDefault = [ ...base.map(([ a, b ]) => [ `${a}`, `${b}Weak` ]) ] as ThemeVarPair[];
+  // const defaultOnStrong = [ ...base.map(([ a, b ]) =>
+  //   [ `${a}Strong`, `${b}` ]) ] as ThemeVarPair[];
```

Comment:
nit: either uncomment, remove, or include a reason for  why it's commented out

Reply:
hmm I didn't include them since these would score very low in our contrast check 🤔  maybe I'll put them in a separate test with lower tolerance

Reply:
done